src/vm/functions/database.rs

```diff
@@ -39,7 +39,7 @@ pub fn special_contract_call(args: &[SymbolicExpression],
         },
         SymbolicExpressionType::Atom(contract_ref) => {
             // Dynamic dispatch
-            match context.callable_contracts.get(contract_ref) {
+            match context.lookup_callable_contract(contract_ref) {
```

Comment:
the code for `contract-of` was modeled after the `contract-call?`, so indeed `contract-call?` would have the same problem.